vm/src/obj/objdict.rs

```diff
@@ -247,6 +234,54 @@ impl ItemProtocol for PyDictRef {
     }
 }
 
+// Implement IntoIterator so that we can easily iterate dictionaries from rust code.
+impl IntoIterator for PyDictRef {
+    type Item = (PyObjectRef, PyObjectRef);
+    type IntoIter = DictIterator;
+
+    fn into_iter(self) -> Self::IntoIter {
+        DictIterator::new(self)
+    }
+}
+
+impl IntoIterator for &PyDictRef {
+    type Item = (PyObjectRef, PyObjectRef);
+    type IntoIter = DictIterator;
+
+    fn into_iter(self) -> Self::IntoIter {
+        DictIterator::new(self.clone())
+    }
+}
+
+pub struct DictIterator {
```

Comment:
Nit - could we call this `DictIter`? (just `Iter` is more common and also serves to distinguish between iterators for use from rust vs from python)